orchagent/intfsorch.cpp

```diff
@@ -466,6 +507,9 @@ bool IntfsOrch::removeRouterIntfs(Port &port)
         return false;
     }
 
+    const auto id = sai_serialize_object_id(port.m_rif_id);
+    removeRifFromFlexCounter(id, port.m_alias);
```

Comment:
Why is this called directly whereas `addRifToFlexCounter `is based on timer?

Reply:
this is due to FC logic.
The Flex counter will try to determine the object's type based on the vid. The VIDTOIRID is not instantly available on create.
For the delete flow it's  the opposite.

If there will be some issues with this logic, I believe the FC flow should be changed. Right now it's tricky to dynamically add/remove objects that need to be polled.

Reply:
How to ensure deleting flex counter before deleting rif in syncd? If deleting rif happened before deleting flex counter, rid cannot be found in the VIDTOIRID.